ksp_plugin/plugin.cpp

```diff
@@ -444,19 +500,28 @@ void Plugin::WriteToMessage(
 
   planetarium_rotation_.WriteToMessage(message->mutable_planetarium_rotation());
   current_time_.WriteToMessage(message->mutable_current_time());
+  history_time_.WriteToMessage(message->mutable_history_time());
   Index const sun_index = FindOrDie(celestial_to_index, sun_);
   message->set_sun_index(sun_index);
+  LOG(INFO) << NAMED(message->ByteSize());
 }
 
 std::unique_ptr<Plugin> Plugin::ReadFromMessage(
     serialization::Plugin const& message) {
   LOG(INFO) << __FUNCTION__;
+  auto ephemeris = Ephemeris<Barycentric>::ReadFromMessage(message.ephemeris());
+  auto const& bodies = ephemeris->bodies();
+  auto bodies_it = bodies.begin();
   IndexToOwnedCelestial celestials;
   for (auto const& celestial_message : message.celestial()) {
-    celestials.emplace(
-        celestial_message.index(),
-        Celestial::ReadFromMessage(celestial_message.celestial()));
```

Comment:
Yes, but that sort of cleanup will get done when we rewrite the plugin for Brouwer.